source/tyr/tests/integration/purge_test.py

```diff
@@ -207,3 +216,34 @@ def test_purge_old_jobs_same_dataset():
 
     folders = set(glob.glob('{}/*'.format(backup_dir)))
     assert len(folders) == 1
+
+
+@pytest.mark.usefixtures("init_instances_dir")
+def test_purge_old_jobs_of_multi_instance():
+    """
+    We should delete jobs, data_sets as well as metrics of all instances
+    including discarded = True
+    """
+    app.config['JOB_MAX_PERIOD_TO_KEEP'] = 1
+
+    instance_name, backup_dir = init_test()
+    create_jobs_with_same_datasets(instance_name, backup_dir)
+
+    create_jobs_with_same_datasets('fr-discarded', backup_dir)
+
+    instances_resp = api_get('/v0/instances')
+    assert len(instances_resp) == 2
+
+    # Delete the second instance (discarded = True)
+    instances_resp = api_delete('/v0/instances/fr-discarded')
+    assert instances_resp['discarded'] is True
+
+    jobs_resp = api_get('/v0/jobs/fr-discarded')
+    assert len(jobs_resp['jobs']) == 3
+    jobs_resp = api_get('/v0/jobs/fr')
+    assert len(jobs_resp['jobs']) == 3
+    tasks.purge_jobs()
+    jobs_resp = api_get('/v0/jobs/fr-discarded')
+    assert len(jobs_resp['jobs']) == 1
```

Comment:
Does this mean that for a discarded instance, the last dataset is still kept?

Reply:
Exactly. We will have only fewer datasets of discarded instance.